code/game/gamemodes/cult/runes.dm

```diff
@@ -1,1043 +1,992 @@
-			ticker.mode:eldergod = 0
-		return
-	else
-		return fizzle()
-
-/////////////////////////////////////////FIFTH RUNE
-/obj/effect/rune/proc/empRune()
-	emp(src.loc,3)
+/*
+There are a few different procs each rune runs through when a cultist activates it.
+can_invoke() is called when a cultist activates the rune with an empty hand. If there are multiple cultists, this rune determines if the required amount is nearby.
+invoke() is the rune's actual effects.
+fail_invoke() is called when the rune fails, via not enough people around or otherwise. Typically this just has a generic 'fizzle' effect.
+structure_check() searches for nearby cultist structures required for the invocation. Proper structures are pylons, forges, archives, and altars.
+*/
 
-/obj/effect/rune/proc/emp(var/U,var/range_red) //range_red - var which determines by which number to reduce the default emp range, U is the source loc, needed because of talisman emps which are held in hand at the moment of using and that apparently messes things up -- Urist
-	if(istype(src,/obj/effect/rune))
-		usr.say("Ta'gh fara[pick("'","`")]qha fel d'amar det!")
-	else
-		usr.whisper("Ta'gh fara[pick("'","`")]qha fel d'amar det!")
-	playsound(U, 'sound/items/Welder2.ogg', 25, 1)
-	var/turf/T = get_turf(U)
-	if(T)
-		T.hotspot_expose(700,125)
-	var/rune = src // detaching the proc - in theory
-	empulse(U, (range_red - 2), range_red)
-	qdel(rune)
-	return
+/obj/effect/rune/proc/can_invoke(var/mob/living/user)
+	//This proc determines if the rune can be invoked at the time. If there are multiple required cultists, it will find all nearby cultists.
+	var/list/invokers = list() //people eligible to invoke the rune
+	var/list/chanters = list() //people who will actually chant the rune when passed to invoke()
+	chanters |= user
+	invokers |= user
+	if(req_cultists > 1 || allow_excess_invokers)
+		for(var/mob/living/L in range(1, src))
+			if(iscultist(L))
+				if(L == user)
+					continue
+				if(ishuman(L))
+					var/mob/living/carbon/human/H = L
+					if((H.disabilities & MUTE) || H.silent)
+						continue
+				if(L.stat)
+					continue
+				invokers |= L
+		if(invokers.len >= req_cultists)
+			if(allow_excess_invokers)
+				chanters |= invokers
+			else
+				invokers -= user
+				shuffle(invokers)
+				for(var/i in 0 to req_cultists)
+					var/L = pick_n_take(invokers)
+					chanters |= L
+	return chanters
+
+/obj/effect/rune/proc/invoke(var/list/invokers)
+	//This proc contains the effects of the rune as well as things that happen afterwards. If you want it to spawn an object and then delete itself, have both here.
+	if(invocation)
+		for(var/M in invokers)
+			var/mob/living/L = M
+			L.say(invocation)
+			L.changeNext_move(CLICK_CD_MELEE)//THIS IS WHY WE CAN'T HAVE NICE THINGS
+	var/oldtransform = transform
+	spawn(0) //animate is a delay, we want to avoid being delayed
+		animate(transform = matrix()*2, alpha = 0, time = 5) //fade out
+		animate(transform = oldtransform, alpha = 255, time = 0)
```

Comment:
this is turned into a proc, on TG
```
/obj/effect/rune/proc/do_invoke_glow()
	var/oldtransform = transform
	spawn(0) //animate is a delay, we want to avoid being delayed
		animate(src, transform = matrix()*2, alpha = 0, time = 5) //fade out
		animate(transform = oldtransform, alpha = 255, time = 0)
```